fuse/eval/metrics/metrics_common.py

```diff
@@ -135,7 +135,10 @@ def collect(self, batch: Dict) -> None:
                 for name, key in self._keys_to_collect.items():
                     value = sample[key]
                     if isinstance(value, torch.Tensor):
-                        value = value.detach().cpu().numpy()
+                        value = value.detach()
```

Comment:
To optimize GPU usage we might wanna move the tensor to the CPU before changing the types:

```python
value = value.detach().cpu()
    if value.dtype == torch.bfloat16:
        value = value.to(torch.float)
value = value.numpy()
```

Just a thought.. It's pretty much negligible